src/benchmark_register.h

```diff
@@ -5,29 +5,103 @@
 
 #include "check.h"
 
+namespace benchmark {
+namespace internal {
```

Comment:
I presumed 'AddRanges' wasn't intended for global scope, please correct me if I'm wrong.